src/status_im/chat/models/images.cljs

```diff
@@ -100,13 +100,15 @@
 
 (re-frame/reg-fx
  ::camera-roll-get-photos
- (fn [num]
+ (fn [[num end-cursor]]
    (permissions/request-permissions
     {:permissions [:read-external-storage]
      :on-allowed  (fn []
-                    (-> (.getPhotos CameraRoll #js {:first num :assetType "Photos" :groupTypes "All"})
-                        (.then #(re-frame/dispatch [:on-camera-roll-get-photos (:edges (types/js->clj %))]))
-                        (.catch #(log/warn "could not get cameraroll photos"))))})))
+                    (-> (if end-cursor
+                          (.getPhotos CameraRoll #js {:first num :after end-cursor :assetType "Photos" :groupTypes "All"})
+                          (.getPhotos CameraRoll #js {:first num :assetType "Photos" :groupTypes "All"}))
+                        (.then #(re-frame/dispatch [:on-camera-roll-get-photos (:edges (types/js->clj %)) (:page_info (types/js->clj %)) end-cursor]))
```

Comment:
You could extract the expensive (?) data converstion of `types/js->clj` into a binding.